### ISSUE TITLE
Lock the letter down at the time of sending

### ISSUE THREAD

**Issue by toolness** [JustFixNYC/tenants2#507]
Currently, it's possible for the user to change their letter of complaint at any time, which makes it very difficult to reproduce months later, e.g. if it's needed in court.  We should instead raise an error when the user tries to change their letter after they've confirmed that it be sent.  While not required, there could be a server-defined time window during which it could be changed before it's locked down (and if we don't want such a window to exist in reality, we could simply set it to zero).

**Comment by toolness** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Lock letter down at time of sending (could be changed later)](https://trello.com/c/9FOeC7sR/5-lock-letter-down-at-time-of-sending-could-be-changed-later)

**Comment by toolness** [comment]
Hmm, so something I just realized is that the HP action flow uses the same `issues` model for storing its issues: that is, when a user fills out an HP action, they can add or remove issues from their letter of complaint.

I knew this might be a problem when I first implemented it but just went with it because MVP and deadlines.  But we should figure out how best to address this.

One way is by saving a copy of the letter of complaint PDF at the time it's sent.  We should probably do this regardless, because we don't want e.g. changes in the letter copy reflected on letters that were sent a long time ago.

Another thing we could do is use something like [django-reversion](https://django-reversion.readthedocs.io/en/stable/) to version the issues, and have the letter of complaint point to the proper revision at the time of letter sending, so that we always have access to it.  (For that matter, we might want to do the same for onboarding info, since presumably some of that could change in the future.)  This would allow all of our data to be machine-readable, so that we could e.g. do comparisions against what a user reported in their LoC vs. what they reported in their HP action.
